starlette/applications.py

```diff
@@ -162,9 +169,9 @@ def add_websocket_route(
         self.router.add_websocket_route(path, route, name=name)
 
     def exception_handler(
-        self, exc_class_or_status_code: typing.Union[int, typing.Type[Exception]]
-    ) -> typing.Callable:
-        def decorator(func: typing.Callable) -> typing.Callable:
+        self, exc_class_or_status_code: _ExcKey
+    ) -> typing.Callable[[_C], _C]:
+        def decorator(func: _C) -> _C:
```

Comment:
```python
@exception_handler
def my_handler(request: Request, exception: Exception) -> None: ...

# Before this change, this will reveal `Callable[..., Any]`, after the change
# the original type hints will be left intact.
reveal_type(my_handler)
```

Reply:
can't this be done with `@functools.wraps`?

Reply:
@adriangb As far as I know `@functools.wraps` has no effect on typing?

Reply:
you are right 😄, it copies the docstrings and somehow the right signature for `help()`:

```python
import functools


def decorator(func):
    @functools.wraps(func)
    def wrapper(a: str, b: int) -> bytes:
        return func(a, b)
    
    return wrapper

@decorator
def func(a: int, b: int) -> int:
    """docstring"""
    return a + b

help(func)  # shows func(a: int, b: int) -> int
```

I assumed it did this by copying `wrapper.__signature__ = func.__signature__` or something, but I guess it does not.